### DISCUSSION TITLE
Is it possible to just use the types

### DISCUSSION THREAD

**Discussion by RobinClowers** [adelsz/pgtyped#500] [Q&A]
I have a pretty unique use case. The app I work on queries https://materialize.com/ over http, rather than making a direct pg connection. It would be great if I could just generate the types from queries and use them in my project, with requiring all the pg client stuff. I would imagine these things are fairly coupled, but I thought it was worth asking. I saw https://github.com/adelsz/pgtyped/issues/394, which is in a similar vein, so it might be useful for other cases as well.

**Comment by fjavierv** [comment]
I have similar need. I am developing a backend and fronted, both using TS. I would like to use the types generated in the backend (with pgtyped) in the front, for that I would be nice to split the current generated files in two parts. The first one would be the base types (used by the frontend), this would not depent on anything. The second file would include the base one to define de PreparedQueries.

**Comment by RobinClowers** [comment]
@fjavierv I ended up using https://kysely.dev/, which worked out really well.
